packages/starlight/__tests__/sidebar/navigation.test.ts

```diff
@@ -84,12 +86,28 @@ describe('getSidebar', () => {
 			        "type": "link",
 			      },
 			      {
-			        "attrs": {},
 			        "badge": undefined,
-			        "href": "/reference/frontmatter/",
-			        "isCurrent": false,
-			        "label": "Frontmatter Reference",
-			        "type": "link",
+			        "collapsed": false,
+			        "entries": [
+			          {
+			            "attrs": {},
+			            "badge": undefined,
+			            "href": "/reference/frontmatter/foo/",
+			            "isCurrent": false,
+			            "label": "Foo",
+			            "type": "link",
+			          },
+			          {
+			            "attrs": {},
+			            "badge": undefined,
+			            "href": "/reference/frontmatter/",
+			            "isCurrent": false,
+			            "label": "Frontmatter Reference",
+			            "type": "link",
+			          },
```

Comment:
good catch! Perhaps this sorting id could be generated when creating the tree. Everytime we go into a tree node recursively we could attach the current object property as a path.  This way not only files but directories can also have a sorting id.

Reply:
Do directories need them? It’s only for this one case I think. But yeah, I think doing it then makes sense!

Reply:
I think it could make a more homogeneous comparison. Since it was not possible to know the directory path, comparisons between directories and files were done by comparing a filename and directory name. But if both were in include their paths it would make a bit easier.

Reply:
On second thought I think It would be a better solution to include a slug property in directories `dir["_slug"]` since index pages defined as index or pages of the parent directory share the same slug.

| Route | slug |
|-------|-------|
| `reference/frontmatter.md` | `reference/frontmatter` |
| `reference/frontmatter/index.md` | `reference/frontmatter` |


Then sorting by slug would be very trivial for both directories and files

| Order | Route/Dir | slug |
|-|-------|-------|
|1| `reference/frontmatter.md` | `reference/frontmatter` |
|2| `reference/frontmatter/bar` | `reference/frontmatter/bar` |
|3| `reference/frontmatter/bar/foo.md` | `reference/frontmatter/bar/foo` |
|4| `reference/frontmatter/fiz` | `reference/frontmatter/fiz` |
|5| `reference/frontmatter/fiz/zoo.md` | `reference/frontmatter/fiz/zoo` |
|6| `reference/frontmatter/foo.md` | `reference/frontmatter/foo` |

**Sidebar view**
```
 reference/
    ├── index.md
    ├── bar/
    │   └── foo.md
    ├── fiz/
    │   └── zoo.md
    └── foo.md
```
This way index are always at the top, even in nested directories.

Reply:
Hmm, maybe? Not opposed to trying!

Reply:
added in f4ee4ed258dc4bb2e7994c31e092dd213a224f10, I need some help with types but everything seem to be working